### DISCUSSION TITLE
`pragma warning disable` for current line

### DISCUSSION THREAD

**Discussion by lonix1** [dotnet/csharplang#5648] [Language Ideas] [answered]
We can do this:
````cs
#pragma warning disable X
SomethingThatTriggersX();
#pragma warning enable X
````

But that tends to be very messy, and reduces readability.

In various tools and languages it's possible to disable warnings per-line, which is much neater. It would be nice to have such functionality in C# too. For example:
````cs
SomethingThatTriggersX();              // pragma warning disable-line X

//or

SomethingThatTriggersX();              #pragma warning disable-line X
````

**Comment by DavidArno** [comment]
Aside from a very minor point that I'd like `#pragma warning disable-line X` to appear before the line in question, not after, I'd love to see this feature implemented.

**Comment by lonix1** [comment]
I see what you mean, even if it's on the same line it's considered "after" the line in question.

I think it's neater (and I'm used to it from other environments), but I guess it's a personal choice. If this can be done at all, it would be nice to have both options to keep everyone happy.

**Comment by jnm2** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/1070
